src/util/pointer_expr.cpp

```diff
@@ -227,13 +227,25 @@ void dereference_exprt::validate(
     "operand");
 }
 
-exprt pointer_in_range_exprt::lower() const
+exprt r_or_w_ok_exprt::lower(const namespacet &ns) const
+{
+  return conjunction(
+    {not_exprt{null_object(pointer())},
+     not_exprt{is_invalid_pointer_exprt{pointer()}},
+     not_exprt{deallocated(pointer(), ns)},
+     not_exprt{dead_object(pointer(), ns)},
+     not_exprt{object_lower_bound(pointer(), nil_exprt())},
+     not_exprt{object_upper_bound(pointer(), size())}});
+}
+
+exprt pointer_in_range_exprt::lower(const namespacet &ns) const
 {
   return and_exprt(
     {same_object(op0(), op1()),
      same_object(op1(), op2()),
```

Comment:
Would these expressions be lowered in an ulterior lowering step ? (just thinking about what happen when passing these through `goto-instrument` multiple times)